src/UI/ImageButton.cpp

```diff
@@ -210,6 +210,8 @@ ImageButton::~ImageButton()
 
 Texture* ImageButton::texture()
 {
+    if(!_enabled) return _textures.at(2);
+    
```

Comment:
Maybe we need some constants for those image indexes. Also, don't forget spaces after control flow statements.